Catalog entity cards using the new BUI don't recognise the legacy "variant" property

### 📜 Issue Labels

- [x] Please familiarize yourself with the issue labels used in this project: [LABELS.md](https://github.com/backstage/backstage/blob/master/LABELS.md)

### 🔎 Search Terms

```plain
Catalog entity cards using legacy variant property
```

### 🗃️ Project Area

Catalog

### 🔗 External Integration

N/A

### 📝 Description & Context

**Context**
I have installed backstage `1.51.2` using `create-app` and copied the plugin I have and working just fine on the version `1.44.2`. The plugin uses various entity cards to display entity relations (like `EntityHasSubcomponentsCard` or `EntityDependencyOfComponentsCard`).

Starting with backstage version `1.49.0`, the cards are using BUI instead of MUI and some of the properties were removed as well (like `variant` or `columns`).
However, the [documentation (section "BREAKING: Entity cards migrated to Backstage UI") says](https://github.com/backstage/backstage/releases/tag/v1.49.0) that - at least for some cards - the legacy properties were preserved so that - based on their presence - the catalog would know to display the card using the previous MUI-based implementation (also, [here](https://github.com/backstage/backstage/blob/a22f3649bfa7d7b5c5c91eeba01e54aa1abd3c31/docs/releases/v1.49.0-changelog.md?plain=1#L51))

```The old variant, columns, and tableOptions props are deprecated but still supported — passing them triggers the legacy MUI rendering.```

[Looking at the code for 1.51.2 for DependsOnComponentsCard](https://github.com/backstage/backstage/blob/v1.51.2/plugins/catalog/src/components/DependsOnComponentsCard/DependsOnComponentsCard.tsx#L64), the switch between the new/default properties and the legacy ones is present in the card definition (exported as `EntityDependsOnComponentsCard` from `plugin.esm.js` file):

```
export function DependsOnComponentsCard(
  props: DependsOnComponentsCardProps | DependsOnComponentsCardLegacyProps,
) {
```

**The problem**
Despite these, my plugin fails with an error:

```
plugins/my-app/src/components/catalog/EntityPage.tsx:394:42 - error TS2322: Type '{ variant: string; }' is not assignable to type 'IntrinsicAttributes & DependsOnComponentsCardProps'.
  Property 'variant' does not exist on type 'IntrinsicAttributes & DependsOnComponentsCardProps'.

394           <EntityDependsOnComponentsCard variant="gridItem" />
```

When looking in my `@backstage/plugin-catalog`  (v `2.0.5`) I see that the `DependsOnComponentsCard` is defined **without the legacy property switch** (in `@backstage/plugin-catalog/dist/index.d.ts`):

```
/** @public */
declare const EntityDependsOnComponentsCard: (props: DependsOnComponentsCardProps) => JSX.Element;

```
This is also visible on the package [available for downloading on npmjs @backstage/plugin-catalog/dist/index.d.ts file, line 752)](https://www.npmjs.com/package/@backstage/plugin-catalog/v/2.0.5?activeTab=code) 

```
declare const EntityDependsOnComponentsCard: (props: DependsOnComponentsCardProps) => JSX.Element;
```

Not sure where this difference is coming from, between the backstage documentation and code - on one hand - and what the node module `plugin-catalog` actually contains after installation - on the other hand. 

Please let me know if I'm doing something wrong or - if i is a bug - how I can work around it.

With many thanks,
Silviu

**Local packages**

```yarn backstage-cli info

OS:   Linux 6.18.33.1-microsoft-standard-WSL2 - linux/x64
node: v22.22.2
yarn: 4.10.3
cli:  0.36.2 (installed)
info module: 0.1.2
backstage:   1.51.2

Dependencies:
  @backstage-community/plugin-entity-validation                    0.13.1
  @backstage-community/plugin-sonarqube                            0.20.1
  @backstage-community/plugin-sonarqube-backend                    0.14.0
  @backstage-community/plugin-sonarqube-react                      0.12.0
  @backstage/app-defaults                                          1.7.8
  @backstage/backend-app-api                                       0.8.0, 0.9.3, 1.7.0
  @backstage/backend-common                                        0.23.3, 0.24.1, 0.25.0
  @backstage/backend-defaults                                      0.13.3, 0.17.2, 0.4.4
  @backstage/backend-dev-utils                                     0.1.7
  @backstage/backend-dynamic-feature-service                       0.2.15
  @backstage/backend-openapi-utils                                 0.3.0, 0.5.5, 0.6.9
  @backstage/backend-plugin-api                                    0.6.21, 0.7.0, 0.8.1, 1.9.1
  @backstage/backend-tasks                                         0.5.27
  @backstage/backend-test-utils                                    0.4.4, 1.11.3
  @backstage/catalog-client                                        1.15.2
  @backstage/catalog-model                                         1.9.0
  @backstage/cli                                                   0.34.6, 0.36.2
  @backstage/cli-common                                            0.1.18, 0.2.2
  @backstage/cli-defaults                                          0.1.2
  @backstage/cli-module-actions                                    0.1.1
  @backstage/cli-module-auth                                       0.1.2
  @backstage/cli-module-build                                      0.1.3
  @backstage/cli-module-config                                     0.1.2
  @backstage/cli-module-github                                     0.1.2
  @backstage/cli-module-info                                       0.1.2
  @backstage/cli-module-lint                                       0.1.2
  @backstage/cli-module-maintenance                                0.1.2
  @backstage/cli-module-migrate                                    0.1.2
  @backstage/cli-module-new                                        0.1.3
  @backstage/cli-module-test-jest                                  0.1.2
  @backstage/cli-module-translations                               0.1.2
  @backstage/cli-node                                              0.2.18, 0.3.2
  @backstage/config                                                1.3.8
  @backstage/config-loader                                         1.10.11
  @backstage/core-app-api                                          1.20.1
  @backstage/core-compat-api                                       0.5.11
  @backstage/core-components                                       0.12.5, 0.13.10, 0.18.10
  @backstage/core-plugin-api                                       1.12.6
  @backstage/dev-utils                                             1.1.23
  @backstage/e2e-test-utils                                        0.1.2
  @backstage/errors                                                1.3.1
  @backstage/eslint-plugin                                         0.2.3, 0.3.0
  @backstage/filter-predicates                                     0.1.3
  @backstage/frontend-app-api                                      0.13.3, 0.14.1, 0.16.3
  @backstage/frontend-defaults                                     0.3.6, 0.5.2
  @backstage/frontend-plugin-api                                   0.12.1, 0.13.4, 0.14.1, 0.15.1, 0.16.2, 0.17.1
  @backstage/frontend-test-utils                                   0.4.5, 0.6.0
  @backstage/integration                                           1.20.1, 2.0.2
  @backstage/integration-aws-node                                  0.1.21, 0.2.0
  @backstage/integration-react                                     1.2.18
  @backstage/module-federation-common                              0.1.4
  @backstage/plugin-api-docs                                       0.13.5
  @backstage/plugin-app                                            0.3.5, 0.4.6
  @backstage/plugin-app-backend                                    0.5.14
  @backstage/plugin-app-node                                       0.1.45
  @backstage/plugin-app-react                                      0.1.0, 0.2.3
  @backstage/plugin-app-visualizer                                 0.2.4
  @backstage/plugin-auth                                           0.1.8
  @backstage/plugin-auth-backend                                   0.29.0
  @backstage/plugin-auth-backend-module-github-provider            0.5.3
  @backstage/plugin-auth-backend-module-guest-provider             0.2.19
  @backstage/plugin-auth-backend-module-microsoft-provider         0.3.15
  @backstage/plugin-auth-node                                      0.4.17, 0.5.6, 0.6.14, 0.7.1
  @backstage/plugin-auth-react                                     0.1.27
  @backstage/plugin-catalog                                        2.0.5
  @backstage/plugin-catalog-backend                                1.32.1, 3.7.1
  @backstage/plugin-catalog-backend-module-logs                    0.1.22
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model 0.2.20
  @backstage/plugin-catalog-backend-module-unprocessed             0.6.12
  @backstage/plugin-catalog-common                                 1.1.10
  @backstage/plugin-catalog-graph                                  0.6.4
  @backstage/plugin-catalog-import                                 0.13.13
  @backstage/plugin-catalog-node                                   1.20.1, 2.2.1
  @backstage/plugin-catalog-react                                  1.21.6, 2.1.4, 3.0.0
  @backstage/plugin-catalog-unprocessed-entities                   0.2.31
  @backstage/plugin-catalog-unprocessed-entities-common            0.0.16
  @backstage/plugin-devtools                                       0.1.39
  @backstage/plugin-devtools-backend                               0.5.17
  @backstage/plugin-devtools-common                                0.1.25
  @backstage/plugin-events-backend                                 0.3.16
  @backstage/plugin-events-node                                    0.3.10, 0.4.22
  @backstage/plugin-kubernetes                                     0.12.19
  @backstage/plugin-kubernetes-backend                             0.21.4
  @backstage/plugin-kubernetes-common                              0.9.12
  @backstage/plugin-kubernetes-node                                0.4.4
  @backstage/plugin-kubernetes-react                               0.5.19
  @backstage/plugin-mcp-actions-backend                            0.1.13
  @backstage/plugin-notifications                                  0.5.17
  @backstage/plugin-notifications-backend                          0.6.5
  @backstage/plugin-notifications-common                           0.2.3
  @backstage/plugin-notifications-node                             0.2.26
  @backstage/plugin-org                                            0.6.49
  @backstage/plugin-permission-backend                             0.7.12
  @backstage/plugin-permission-backend-module-allow-all-policy     0.2.19
  @backstage/plugin-permission-common                              0.7.14, 0.8.4, 0.9.9
  @backstage/plugin-permission-node                                0.10.12, 0.11.0, 0.8.8, 0.9.1
  @backstage/plugin-permission-react                               0.4.41, 0.5.1
  @backstage/plugin-proxy-backend                                  0.6.13
  @backstage/plugin-proxy-node                                     0.1.15
  @backstage/plugin-scaffolder                                     1.37.0
  @backstage/plugin-scaffolder-backend                             4.0.0
  @backstage/plugin-scaffolder-backend-module-bitbucket-server     0.2.21
  @backstage/plugin-scaffolder-backend-module-github               0.9.9
  @backstage/plugin-scaffolder-backend-module-notifications        0.1.22
  @backstage/plugin-scaffolder-common                              1.7.6, 2.2.0
  @backstage/plugin-scaffolder-node                                0.12.5, 0.13.3, 0.4.12
  @backstage/plugin-scaffolder-node-test-utils                     0.3.11
  @backstage/plugin-scaffolder-react                               2.0.0
  @backstage/plugin-search                                         1.7.4
  @backstage/plugin-search-backend                                 2.1.2
  @backstage/plugin-search-backend-module-catalog                  0.3.15
  @backstage/plugin-search-backend-module-pg                       0.5.55
  @backstage/plugin-search-backend-module-techdocs                 0.4.14
  @backstage/plugin-search-backend-node                            1.4.4
  @backstage/plugin-search-common                                  1.2.24
  @backstage/plugin-search-react                                   1.11.4
  @backstage/plugin-signals                                        0.0.24
  @backstage/plugin-signals-backend                                0.3.15
  @backstage/plugin-signals-node                                   0.2.1
  @backstage/plugin-signals-react                                  0.0.16, 0.0.22
  @backstage/plugin-techdocs                                       1.17.6
  @backstage/plugin-techdocs-backend                               2.2.0
  @backstage/plugin-techdocs-common                                0.1.1
  @backstage/plugin-techdocs-module-addons-contrib                 1.1.36
  @backstage/plugin-techdocs-node                                  1.15.0
  @backstage/plugin-techdocs-react                                 1.3.11
  @backstage/plugin-user-settings                                  0.9.3
  @backstage/plugin-user-settings-common                           0.1.0
  @backstage/release-manifests                                     0.0.13
  @backstage/test-utils                                            1.7.18
  @backstage/theme                                                 0.2.19, 0.4.4, 0.5.7, 0.7.3
  @backstage/types                                                 1.2.2
  @backstage/ui                                                    0.13.3, 0.14.3, 0.15.0, 0.8.2
  @backstage/version-bridge                                        1.0.12
  @dweber019/backstage-plugin-endoflife                            0.1.2
  @janus-idp/backstage-plugin-analytics-module-matomo              1.8.1
  @janus-idp/backstage-plugin-feedback-backend                     1.7.2
  @k-phoen/backstage-plugin-grafana                                0.1.22
  @roadiehq/scaffolder-backend-module-http-request                 5.7.0
  @roadiehq/scaffolder-backend-module-utils                        4.1.2
  @rsc-labs/backstage-changelog-plugin                             0.5.0
  @rsc-labs/backstage-changelog-plugin-backend                     0.4.0
```

### 👍 Expected Behavior

When using `variant` property for `EntityHasSubcomponentsCard`, `EntityDependencyOfComponentsCard`, `EntityDependsOnComponentsCard`, `EntityHasComponentsCard` and `EntityHasSystemsCard`, the property should be accepted by the catalog and the card to be displayed using MUI.

### 📦 Reproduction Repo

_No response_

### 🥾 Reproduction steps

1. Install (or use) a backstage version 1.51.2
2. Add a card to an entity route in `EntiytPage.tsx` page within a plugin (e.g.., in `plugins/src/components/catalog/EntityPage.tsx`)

```    
    <EntityLayout.Route path="/comp-dependencies" title="Component Dependencies">
      <Grid container spacing={3} alignItems="stretch">
            <Grid item md={6}>
              <EntityDependsOnComponentsCard variant="gridItem"/>
           </Grid>
      </Grid>
    </EntityLayout.Route>
```
3. Start the application

### Have you read the Code of Conduct?

- [x] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Undecided

--- a/.changeset/fix-catalog-legacy-props.md
+++ b/.changeset/fix-catalog-legacy-props.md
@@ -1,0 +1,5 @@
+---
+'@backstage/plugin-catalog': patch
+---
+
+Add support for legacy props in catalog entity card exports by widening the public TypeScript prop types. This ensures compatibility for consumers still using legacy properties such as `variant` and `columns`.

--- a/plugins/catalog/report.api.md
+++ b/plugins/catalog/report.api.md
@@ -504,41 +504,45 @@ export type EntityContextMenuClassKey = 'button';
 
 // @public (undocumented)
 export const EntityDependencyOfComponentsCard: (
-  props: DependencyOfComponentsCardProps,
+  props:
+    | DependencyOfComponentsCardProps
+    | DependencyOfComponentsCardLegacyProps,
 ) => JSX.Element;
 
 // @public (undocumented)
 export const EntityDependsOnComponentsCard: (
-  props: DependsOnComponentsCardProps,
+  props: DependsOnComponentsCardProps | DependsOnComponentsCardLegacyProps,
 ) => JSX.Element;
 
 // @public (undocumented)
 export const EntityDependsOnResourcesCard: (
-  props: DependsOnResourcesCardProps,
+  props: DependsOnResourcesCardProps | DependsOnResourcesCardLegacyProps,
 ) => JSX.Element;
 
 // @public (undocumented)
 export const EntityHasComponentsCard: (
-  props: HasComponentsCardProps,
+  props: HasComponentsCardProps | HasComponentsCardLegacyProps,
 ) => JSX.Element;
 
 // @public (undocumented)
 export const EntityHasResourcesCard: (
-  props: HasResourcesCardProps,
+  props: HasResourcesCardProps | HasResourcesCardLegacyProps,
 ) => JSX.Element;
 
 // @public (undocumented)
 export const EntityHasSubcomponentsCard: (
-  props: HasSubcomponentsCardProps,
+  props: HasSubcomponentsCardProps | HasSubcomponentsCardLegacyProps,
 ) => JSX.Element;
 
 // @public (undocumented)
 export const EntityHasSubdomainsCard: (
-  props: HasSubdomainsCardProps,
+  props: HasSubdomainsCardProps | HasSubdomainsCardLegacyProps,
 ) => JSX.Element;
 
 // @public (undocumented)
-export const EntityHasSystemsCard: (props: HasSystemsCardProps) => JSX.Element;
+export const EntityHasSystemsCard: (
+  props: HasSystemsCardProps | HasSystemsCardLegacyProps,
+) => JSX.Element;
 
 // @public (undocumented)
 export const EntityLabelsCard: (props: EntityLabelsCardProps) => JSX_2.Element;

--- a/plugins/catalog/src/plugin.ts
+++ b/plugins/catalog/src/plugin.ts
@@ -44,14 +44,38 @@ import {
 } from '@backstage/plugin-search-react';
 import { DefaultStarredEntitiesApi } from './apis';
 import { DefaultCatalogPageProps } from './components/CatalogPage';
-import { DependencyOfComponentsCardProps } from './components/DependencyOfComponentsCard';
-import { DependsOnComponentsCardProps } from './components/DependsOnComponentsCard';
-import { DependsOnResourcesCardProps } from './components/DependsOnResourcesCard';
-import { HasComponentsCardProps } from './components/HasComponentsCard';
-import { HasResourcesCardProps } from './components/HasResourcesCard';
-import { HasSubcomponentsCardProps } from './components/HasSubcomponentsCard';
-import { HasSubdomainsCardProps } from './components/HasSubdomainsCard';
-import { HasSystemsCardProps } from './components/HasSystemsCard';
+import {
+  DependencyOfComponentsCardLegacyProps,
+  DependencyOfComponentsCardProps,
+} from './components/DependencyOfComponentsCard';
+import {
+  DependsOnComponentsCardLegacyProps,
+  DependsOnComponentsCardProps,
+} from './components/DependsOnComponentsCard';
+import {
+  DependsOnResourcesCardLegacyProps,
+  DependsOnResourcesCardProps,
+} from './components/DependsOnResourcesCard';
+import {
+  HasComponentsCardLegacyProps,
+  HasComponentsCardProps,
+} from './components/HasComponentsCard';
+import {
+  HasResourcesCardLegacyProps,
+  HasResourcesCardProps,
+} from './components/HasResourcesCard';
+import {
+  HasSubcomponentsCardLegacyProps,
+  HasSubcomponentsCardProps,
+} from './components/HasSubcomponentsCard';
+import {
+  HasSubdomainsCardLegacyProps,
+  HasSubdomainsCardProps,
+} from './components/HasSubdomainsCard';
+import {
+  HasSystemsCardLegacyProps,
+  HasSystemsCardProps,
+} from './components/HasSystemsCard';
 import { RelatedEntitiesCardProps } from './components/RelatedEntitiesCard';
 import { CatalogSearchResultListItemProps } from './components/CatalogSearchResultListItem';
 import { DefaultEntityPresentationApi } from './apis/EntityPresentationApi';
@@ -159,20 +183,21 @@ export const EntityLabelsCard = catalogPlugin.provide(
 );
 
 /** @public */
-export const EntityHasSystemsCard: (props: HasSystemsCardProps) => JSX.Element =
-  catalogPlugin.provide(
-    createComponentExtension({
-      name: 'EntityHasSystemsCard',
-      component: {
-        lazy: () =>
-          import('./components/HasSystemsCard').then(m => m.HasSystemsCard),
-      },
-    }),
-  );
+export const EntityHasSystemsCard: (
+  props: HasSystemsCardProps | HasSystemsCardLegacyProps,
+) => JSX.Element = catalogPlugin.provide(
+  createComponentExtension({
+    name: 'EntityHasSystemsCard',
+    component: {
+      lazy: () =>
+        import('./components/HasSystemsCard').then(m => m.HasSystemsCard),
+    },
+  }),
+);
 
 /** @public */
 export const EntityHasComponentsCard: (
-  props: HasComponentsCardProps,
+  props: HasComponentsCardProps | HasComponentsCardLegacyProps,
 ) => JSX.Element = catalogPlugin.provide(
   createComponentExtension({
     name: 'EntityHasComponentsCard',
@@ -185,7 +210,7 @@ export const EntityHasComponentsCard: (
 
 /** @public */
 export const EntityHasSubcomponentsCard: (
-  props: HasSubcomponentsCardProps,
+  props: HasSubcomponentsCardProps | HasSubcomponentsCardLegacyProps,
 ) => JSX.Element = catalogPlugin.provide(
   createComponentExtension({
     name: 'EntityHasSubcomponentsCard',
@@ -200,7 +225,7 @@ export const EntityHasSubcomponentsCard: (
 
 /** @public */
 export const EntityHasSubdomainsCard: (
-  props: HasSubdomainsCardProps,
+  props: HasSubdomainsCardProps | HasSubdomainsCardLegacyProps,
 ) => JSX.Element = catalogPlugin.provide(
   createComponentExtension({
     name: 'EntityHasSubdomainsCard',
@@ -213,7 +238,7 @@ export const EntityHasSubdomainsCard: (
 
 /** @public */
 export const EntityHasResourcesCard: (
-  props: HasResourcesCardProps,
+  props: HasResourcesCardProps | HasResourcesCardLegacyProps,
 ) => JSX.Element = catalogPlugin.provide(
   createComponentExtension({
     name: 'EntityHasResourcesCard',
@@ -226,7 +251,7 @@ export const EntityHasResourcesCard: (
 
 /** @public */
 export const EntityDependsOnComponentsCard: (
-  props: DependsOnComponentsCardProps,
+  props: DependsOnComponentsCardProps | DependsOnComponentsCardLegacyProps,
 ) => JSX.Element = catalogPlugin.provide(
   createComponentExtension({
     name: 'EntityDependsOnComponentsCard',
@@ -241,7 +266,9 @@ export const EntityDependsOnComponentsCard: (
 
 /** @public */
 export const EntityDependencyOfComponentsCard: (
-  props: DependencyOfComponentsCardProps,
+  props:
+    | DependencyOfComponentsCardProps
+    | DependencyOfComponentsCardLegacyProps,
 ) => JSX.Element = catalogPlugin.provide(
   createComponentExtension({
     name: 'EntityDependencyOfComponentsCard',
@@ -256,7 +283,7 @@ export const EntityDependencyOfComponentsCard: (
 
 /** @public */
 export const EntityDependsOnResourcesCard: (
-  props: DependsOnResourcesCardProps,
+  props: DependsOnResourcesCardProps | DependsOnResourcesCardLegacyProps,
 ) => JSX.Element = catalogPlugin.provide(
   createComponentExtension({
     name: 'EntityDependsOnResourcesCard',

